packages/dapp-connector/src/connector.js

```diff
@@ -0,0 +1,140 @@
+      promisesMap.set(requestId, {resolve, reject})
+    })
+    const params = {args, browserContext: getContext()}
+    postMessage({id: requestId, method, params})
+    return promise
+  }
+
+  const logMessage = (...args) => {
+    callExternalMethod('log_message', args, {doNotWaitForResponse: true})
+  }
+
+  window.addEventListener('error', (event) => {
+    logMessage('Unhandled error:' + serializeError(event.error))
+  })
+
+  window.addEventListener('unhandledrejection', (event) => {
+    logMessage('Unhandled rejection:' + serializeError(event.reason))
+  })
+
+  const serializeError = (error) => {
+    if (error instanceof Error) {
+      return error.name + ': ' + error.message
+    }
+    return JSON.stringify(error)
+  }
+
+  window.addEventListener('message', (event) => {
+    logMessage('Received message' + JSON.stringify(event.data))
+    const {id, result, error} = event.data
+    const promise = promisesMap.get(id)
+    if (!promise) return
+    promisesMap.delete(id)
+    if (error) {
+      promise.reject(handleError(error))
+    } else {
+      promise.resolve(result)
+    }
+  })
+
+  const createApi = (cardanoEnableResponse) => {
+    if (!cardanoEnableResponse) {
+      logMessage('User Rejected')
+      throw {code: -3, info: 'User Rejected'}
+    }
+    localStorage.setItem('yoroi-session-id', sessionId)
+
+    enabling = false
+    return {
+      getBalance: (...args) => callExternalMethod('api.getBalance', args),
+      getChangeAddress: (...args) => callExternalMethod('api.getChangeAddress', args),
+      getNetworkId: (...args) => callExternalMethod('api.getNetworkId', args),
+      getRewardAddresses: (...args) => callExternalMethod('api.getRewardAddresses', args),
+      getUsedAddresses: (...args) => callExternalMethod('api.getUsedAddresses', args),
+    }
+  }
+
+  const getContext = () => {
+    return {origin: window.location.origin}
+  }
+
+  let enabling = false
+  const isEnabled = () => {
+    if (localStorage.getItem('yoroi-session-id') !== sessionId) {
+      enabling = true
+      logMessage('Account Change')
+      throw {code: -4, info: 'Account Change'}
+    }
+
+    if (enabling) {
+      return false
+    }
+
+    return callExternalMethod('cardano_is_enabled')
+  }
+
+  const enable = (...args) => {
+    enabling = true
+    localStorage.setItem('yoroi-session-id', sessionId)
+    return callExternalMethod('cardano_enable', args).then(createApi).catch(handleError)
+  }
+
+  const handleError = (error) => {
+    if (error.message.toLowerCase().includes('user rejected')) {
+      logMessage('User Rejected')
+      return {code: -3, info: 'User Rejected'}
+    }
+    logMessage('Error:' + error.message)
+    return {code: -1, info: error.message}
+  }
+
+  const walletObj = Object.freeze({
+    enable,
+    isEnabled,
+    icon: iconUrl,
+    apiVersion: apiVersion,
+    name: walletName,
+    supportedExtensions: Object.freeze(supportedExtensions),
+  })
+
+  window.cardano = window.cardano || {}
```

Comment:
prefer ?? for defaulting
```suggestion
  window.cardano = window.cardano ?? {}
```